src/auth/authRegistry.ts

```diff
@@ -0,0 +1,32 @@
+import {AuthData, AuthError} from './types';
+import PolkadotAuth from './polkadotAuth';
+import EthAuth from './ethAuth';
+import SolanaAuth from './solanaAuth';
+
+const _ = require('lodash');
+
+const mapBySigType = (sigTypes: string[], authObject: object) => {
+  return _.zipObject(sigTypes, _.fill(Array(_.size(sigTypes)), authObject));
+};
+
+const authProviders = {
+  ...mapBySigType(['polkadot', 'polka', 'substrate', 'crust'], PolkadotAuth),
```

Comment:
I think this should be all start with 3-chars(like token name) or full-name, that's more fit for Web3 user's habbits